lightningd/lightningd.c

```diff
@@ -234,6 +236,26 @@ static void init_txfilter(struct wallet *w, struct txfilter *filter)
 	}
 }
 
+static void pidfile_create(const struct lightningd *ld)
+{
+        int pid_fd;
+
+        /* Create PID file */
+        pid_fd = open(ld->pidfile, O_RDWR|O_CREAT, 0640);
+        if (pid_fd < 0)
+                errx(1, "Failed to open PID file: %s", strerror(errno));
+        if (lockf(pid_fd, F_TLOCK, 0) < 0)
+                /* Can't lock file */
+                errx(1, "Cannot lock PID file: %s", strerror(errno));
+        char pid[256];
+        /* Get current PID */
+        sprintf(pid, "%d\n", getpid());
+        /* Write PID to lockfile */
+        if (write(pid_fd, pid, strlen(pid)) < 0)
```

Comment:
Use `write_all` from `<ccan/read_write_all/read_write_all.h>` instead to ensure writing all of the data.